package.json

```diff
@@ -18,6 +18,7 @@
   "dependencies": {
     "async": "^2.5.0",
     "async-lock": "^1.1.2",
```

Comment:
Looks like async-lock was previously added and removed for `get_best_network_hash` and is now unused. Adding locks adds new sets of complexities to deal with like deadlocks. I haven't needed to add any locking code for javascript even with async functions given the single-threaded concurrent-execution nature of the language.

So at minimum there needs to be a test to actually fail in a way that have shown up in logs, and from there we can see if locks are actually necessary.

Reply:
I think the async-lock at some point was a (maybe questionable) way to avoid the node.js server from simultaneously launching a lot of clients that end up doing very expensive operations.